### PR TITLE
Upgrade to Swagger UI 3.0.21

### PR DESCRIPTION
**Pull request by carlmon**
- `BooleanValues` and `JsonEditor` configuration options are no longer supported.
- All other SB configuration options ported to 3.0.21 index.html template.
- No new configuration options introduced.

### CODE REVIEW

**Comment by carlmon** [comment]
The failing tests are not due to this PR

**Comment by bjornharrtell** [comment]
Nice work @carlmon. I would think https://github.com/domaindrivendev/Swashbuckle/blob/master/Swashbuckle.Core/Application/SwaggerUiConfig.cs needed some updated too but perhaps not?

**Comment by carlmon** [comment]
Thanks @bjornharrtell, I will update it. Do you prefer [Obsolete] on the unsupported options? Or should I remove the options? I will also update the usage documentation.

**Comment by bjornharrtell** [comment]
Not my call but I would prefer `[Obsolete]` to (hopefully) make this a non-breaking upgrade.

File: src/Swashbuckle.AspNetCore.SwaggerUI/Template/index.html [hunk 1]
```diff
@@ -1,144 +1,129 @@
...
-<div id='header'>
-  <div class="swagger-ui-wrap">
-    <a id="logo" href="http://swagger.io"><img class="logo__img" alt="swagger" height="30" width="30" src="images/logo_small.png" /><span class="logo__title">swagger</span></a>
-    <form id='api_selector'>
-      <div class='input'>
-        <input placeholder="http://example.com/api" id="input_baseUrl" name="baseUrl" type="text" />
-        <div id='auth_container'></div>
-        <select id="select_document"></select>
-        <script type="text/javascript">
-          // Append configured endpoints as options in the document selector
```
Comment (dennismaxjung) [inline src/Swashbuckle.AspNetCore.SwaggerUI/Template/index.html]: You removed the "select document" - dropdown. 
Know you can add multiple swagger endpoint but you cant switch between them on the UI.

File: src/Swashbuckle.AspNetCore.SwaggerUI/Template/index.html [hunk 2]
```diff
@@ -1,144 +1,129 @@
...
-          }
-
-          _.each(JSConfig.OnCompleteScripts, function (script) {
-            $.getScript(script);
-          });
-        },
-        onFailure: function(data) {
-          log("Unable to Load SwaggerUI");
-
-          _.each(JSConfig.OnFailureScripts, function (script) {
```
Comment (dennismaxjung) [inline src/Swashbuckle.AspNetCore.SwaggerUI/Template/index.html]: Throwing "Uncaught ReferenceError: _ is not defined"

File: src/Swashbuckle.AspNetCore.SwaggerUI/Template/index.html [hunk 3]
```diff
@@ -1,144 +1,129 @@
...
+
+  function log() {
+    if ('console' in window) {
+      console.log.apply(console, arguments);
+    }
+  }
+  
+  // Build a system
+  const ui = SwaggerUIBundle({
+    url: JSConfig.SwaggerEndpoints[0].Url,
```
Comment (dennismaxjung) [inline src/Swashbuckle.AspNetCore.SwaggerUI/Template/index.html]: Change url to urls, (see: https://github.com/swagger-api/swagger-ui/blob/master/README.md#parameters) and change the SwaggerEndpoint object to an array of API definition objects {url: "<url>", name: "<name>"}

Comment (carlmon) [inline src/Swashbuckle.AspNetCore.SwaggerUI/Template/index.html]: Nice catch, thanks Dennis!

I updated the PR to support both cases of single and multiple endpoints.

File: src/Swashbuckle.AspNetCore.SwaggerUI/Template/index.html (hunk 2 above)
Comment (carlmon) [inline src/Swashbuckle.AspNetCore.SwaggerUI/Template/index.html]: Thanks @dennismaxjung! I fixed this and updated the PR

**Comment by dennismaxjung** [comment]
It's working, but there seems to be some error at the build task. Some NuGet versions conflicts

**Comment by heldersepu** [comment]
Has anyone tested OAuth2 with this change?

**Comment by cfeilen** [comment]
@dennismaxjung, I was able to fix the nuget errors by upgrading the Microsoft.NETCore.Platforms package from 1.0.2 to 1.1.0,
Fix is in test\Swashbuckle.AspNetCore.IntegrationTests\Swashbuckle.AspNetCore.IntegrationTests.csproj, and test\Swashbuckle.AspNetCore.SwaggerGen.Test\Swashbuckle.AspNetCore.SwaggerGen.Test.csproj.

Issue most likely stems from the fact that Appvyor has probably updated to .NETCore2.0, which now respects the Nuget errors (specifically NU1605)

**Comment by dennismaxjung** [comment]
@cfeilen yes that seems to be the problem.
Maybe we could allow to disable error Check at ui by setting this option  not to the default value if it set to null

**Comment by HideDev** [comment]
Hi, any news?

**Comment by wallymathieu** [comment]
I've added a pull request to fix the build failure in the branch: https://github.com/GenasysTechnologies/Swashbuckle.AspNetCore/pull/1

**Comment by domaindrivendev** [comment]
I def want to support 3x but am also concerned that it's going to be a significant change for folks who've become used to the 2x UI. So, for now, I think it might be better to make it opt-in. For this, I would suggest creating a new project in the repo called Swashbuckle.AspNetCore.SwaggerUI3. Then, opt-in would be supported by installing the additional package and then enabling the 3x UI as follows:

```csharp
app.UseSwaggerUI3(c =>
{
    ...
});
```

Also, I'd like to ensure maintain feature parity if possible - with this PR the MultipleVersions and OAuth2Integration examples don't appear to be working.

**Comment by wallymathieu** [comment]
That sounds nicer

**Comment by dennismaxjung** [comment]
@domaindrivendev
You're right. This might be the best solution for keeping the old v2 and providing the new v3.

Maybe the goal should be to share as much code as possible between both projects? 

Great would be to allow the user easy switching between them.

**Comment by dfoulk** [comment]
Has SwaggerUI 3 been added yet? I noticed a pull request by @alexvaluyskiy, but it doesn't appear to be merged...

**Comment by domaindrivendev** [comment]
See #543

**Pull request closed without merging**